src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,319 @@
+        T apply(Jedis jedis) throws IOException;
+    }
+
+    private <T> T withRedis(RedisCallback<T> callback) throws IOException {
+        Objects.requireNonNull(redisConnectionPool, "redis connection pool must be initialized");
+        Objects.requireNonNull(callback, "callback needs to be provided");
+        try(Jedis jedis = redisConnectionPool.getResource()){
+            try {
+                T result = callback.apply(jedis);
+                redisConnectionPool.returnResourceObject(jedis);
+                return result;
+            } catch(IOException ioe) {
+                redisConnectionPool.returnBrokenResource(jedis);
+                throw ioe;
+            }
+        }catch(Exception e) {
+            throw new RuntimeException("Error working with Redis", e);
+        }
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // END REDIS
+    //------------------------------------------------------------------------------------------------------------------
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START Lifecycle
+    //------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    protected synchronized void startInternal() throws LifecycleException {
+        super.startInternal();
+        setState(LifecycleState.STARTING);
+        redisConnectionPoolConfig = new JedisPoolConfig();
+        redisConnectionPool = new JedisPool(this.redisConnectionPoolConfig, dbHost, dbPort, dbTimeout, dbPassword);
+
+        dateFormat = new SimpleDateFormat(ISO_DATE_FORMAT);
+    }
+
+    @Override
+    protected synchronized void stopInternal() throws LifecycleException {
+        setState(LifecycleState.STOPPING);
+        try {
+            redisConnectionPool.destroy();
+        } catch(Exception e) {
+            log.error("Error - Context: stopInternal. Description: " + e.getMessage());
+        }
+        super.stopInternal();
+    }
+
+    /**
+     * Add a lifecycle event listener to this component.
+     *
+     * @param listener The listener to add
+     */
+    @Override
+    public void addLifecycleListener(LifecycleListener listener) {
+        lifecycle.addLifecycleListener(listener);
+    }
+
+    /**
+     * Get the lifecycle listeners associated with this lifecycle. If this
+     * Lifecycle has no listeners registered, a zero-length array is returned.
+     */
+    @Override
+    public LifecycleListener[] findLifecycleListeners() {
+        return lifecycle.findLifecycleListeners();
+    }
+
+
+    /**
+     * Remove a lifecycle event listener from this component.
+     *
+     * @param listener The listener to remove
+     */
+    @Override
+    public void removeLifecycleListener(LifecycleListener listener) {
+        lifecycle.removeLifecycleListener(listener);
+    }
```

Comment:
You don't have to override everything. Let your base class do some heavy lifting for you.